torch/CMakeLists.txt

```diff
@@ -356,6 +356,12 @@ if(USE_UCC)
   target_compile_definitions(torch_python PRIVATE USE_UCC)
 endif()
 
+if(USE_OPENTELEMTRY)
+  find_package(opentelemetry-cpp CONFIG REQUIRED)
+  target_include_directories(torch_python PRIVATE ${OPENTELEMETRY_CPP_INCLUDE_DIRS})
+  target_link_libraries(torch_python PRIVATE ${OPENTELEMETRY_CPP_LIBRARIES})
```

Comment:
Does this refer only to the otel-api part we discussed or is this the whole library for now?

Reply:
It is the whole library for now and will definitely need to be fixed up.
I'm testing providing the following option `WITH_API_ONLY` as seen in the CMakeLists.txt file[ here](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/CMakeLists.txt#L268).
The unfortunate thing is that for now, opentelemetry have only easily provided a way of incorporating both API + SDK in an existing project. 
See the note [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md#incorporating-into-an-existing-cmake-project).
But, I hope it will be possible for us to split this up.

Reply:
The splitting is done. I was able to build some tests with `WITH_API_ONLY` option.